src/log/log.c

```diff
@@ -1364,7 +1366,7 @@ __wti_log_acquire(WT_SESSION_IMPL *session, uint64_t recsize, WTI_LOGSLOT *slot)
     if (F_ISSET(log, WTI_LOG_FORCE_NEWFILE) || !__log_size_fit(session, &log->alloc_lsn, recsize)) {
         WT_RET(__log_newfile(session, false, &created_log));
         F_CLR(log, WTI_LOG_FORCE_NEWFILE);
-        if (log->log_close_fh != NULL)
+        if (__wt_atomic_load_pointer(&log->log_close_fh) != NULL)
```

Comment:
My understanding is that this branch is trying to check whether `__log_newfile` have set new `log->log_close_fh` on the line before. If that's right, the dangerous of concurrent access here could be that `__log_file_server` can read the new value of   `log->log_close_fh`, close it, and then store `NULL` there even before we check it in this place.

So if my understanding is correct, I would propose to somehow return the information whether new `log_close_fh` was stored as a part of `__log_newfile` or not. One way could be using `log->log_fh` instead of `log->log_close_fh`, and we already read it without any atomic acesses inside `__log_newfile`, however I see that we store it with the release semantic there, so it could be worth checking as well

Reply:
Damn, i understand the predicament here. I actually don't know why I did this in the first place. I removed it and no new warnings appeared. For now I will remove it, let's focus on fixing harder warnings like these in future tickets.